File: app/src/main/java/com/example/bghub/Views/Profile/ProfileFragment.java
package com.example.bghub.Views.Profile;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toolbar;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.bghub.R;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class ProfileFragment extends DaggerFragment implements ProfileContract.View {

    @Inject
    ProfileContract.Presenter mPresenter;

    private static final String TAG = "ProfileFragment";

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        View root = inflater.inflate(R.layout.fragment_profile, container, false);

        return root;
    }


}

File: app/src/main/java/com/example/bghub/Views/Fragments/ProfileFragment.kt
package com.example.bghub.Views.Fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SearchView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.bghub.R
import com.example.bghub.Repositories.Data.DataContract

class ProfileFragment : Fragment() {

    lateinit var mDataRepository: DataContract.Repository

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_profile, container, false)
    }

    fun setDataRepository(dataRepository : DataContract.Repository) {
        mDataRepository = dataRepository;
    }

    companion object {
        @JvmStatic
        fun newInstance(dataRepository : DataContract.Repository): ProfileFragment {
            val fragment = ProfileFragment()
            fragment.setDataRepository(dataRepository)
            return fragment
        }
    }


}